contrib/babelfishpg_tsql/sql/upgrades/babelfishpg_tsql--4.4.0--5.0.0.sql

```diff
@@ -187,12 +187,35 @@ UPDATE sys.babelfish_authid_login_ext SET is_fixed_role = 1 WHERE rolname = 'sys
 
 -- At this point, there will be only one database fixed role which is db_owner.
 UPDATE sys.babelfish_authid_user_ext SET is_fixed_role = 1 WHERE orig_username = 'db_owner';
-UPDATE sys.babelfish_authid_user_ext SET is_fixed_role = 0 WHERE orig_username != 'db_owner';
 
 CREATE OR REPLACE PROCEDURE sys.create_db_roles_during_upgrade()
 LANGUAGE C
 AS 'babelfishpg_tsql', 'create_db_roles_during_upgrade';
-CALL sys.create_db_roles_during_upgrade();
+
+DO $$
+DECLARE
+    role_info record;
+BEGIN
+    WITH fixed_db_roles AS (
+        SELECT DISTINCT orig_username
+        FROM sys.babelfish_authid_user_ext
+        WHERE is_fixed_role = 1
+          AND orig_username IN ('db_owner', 'db_securityadmin', 'db_accessadmin', 
+                                'db_datareader', 'db_datawriter', 'db_ddladmin')
+    )
+    SELECT 
+        string_agg(orig_username, ', ') AS existing_db_roles,
+        COUNT(*) AS role_count
+    INTO role_info
+    FROM fixed_db_roles;
+
+    IF role_info.role_count = 1 THEN
+        CALL sys.create_db_roles_during_upgrade();
+    ELSE
+        RAISE WARNING 'The following fixed database role(s) already exist(s): %', role_info.existing_db_roles;
+    END IF;
```

Comment:
I think `role_count = 5` would be a more strict condition (excluding db_owner), plus lets RAISE NOTICE instead of warning.

Reply:
we want to call the proc if only is_fixed role count is 1 the ele condition will fall down to role_count = 6 since all the other roles are created at once.